python/lsst/daf/persistence/storageInterface.py

```diff
@@ -64,28 +65,69 @@ def registerFormatter(cls, formatable, formatter):
             The type of StorageInterface the formatter is being registered for.
         formatable : class object
             The class object whose instances can be formatted by the formatter.
-        formatter : Formatter class
-            The formatter class that can be used by the StorageInterface instance to read and write the object
-            to the storage.
+        readFormatter : a read formatter callable
+            The formatter class that can be used by the StorageInterface instance to read the object from the
+            storage.
+        writeFormatter : a write formatter callable
+            The formatter class that can be used by the StorageInterface instance to write the object to the
+            storage.
 
         Raises
         ------
         RuntimeError
-            For each object type and StorageInterface subclass a formatter should only be registered once. If
-            a second registration occurs a RuntimeError is raised.
+            For each object type and StorageInterface subclass the read and write formatters should only be
+            registered once. If a second registration occurs for either a RuntimeError is raised.
         """
-        classFormatters = StorageInterface.formatters.setdefault(cls, {})
+        cls.registerReadFormatter(formatable, readFormatter)
+        cls.registerWriteFormatter(formatable, writeFormatter)
+
+    @classmethod
+    def registerReadFormatter(cls, formatable, readFormatter):
+        """Similar to registerFormatters, but only for the readFormatter."""
+        cls._registerFormatter(formatable, readFormatter, 'read')
+
+    @classmethod
+    def registerWriteFormatter(cls, formatable, writeFormatter):
```

Comment:
Instead three separate methods `registerFormatters`, `registerReadFormatter`, `registerWriteFormatter` I'd probably define just one with keyword arguments:

    def registerFormatters(cls, formattable, read=None, write=None)
        ....

Feels more Pythonic to me :snake: